w3-hw1-ThuyDung/cypress/e2e/formPage.cy.js

```diff
@@ -0,0 +1,274 @@
+import { formPage } from "../pages/formPage";
+
+describe('User can register', () => {
+  beforeEach(() => {
+    cy.fixture("data").as("data");
+    cy.visit(Cypress.env("form"));
+  });
+  it("TC-FORM-05: Verify that user can registers when user inputs all fields with valid data ", () => {
+    cy.get("@data").then((data) => {
+      formPage
+        .typeFirstName(data.profiles[5].firstname)
+        .typeLastName(data.profiles[5].lastname)
+        .typeEmail(data.profiles[5].email)
+        .selectGender(data.profiles[5].gender)
+        .typeMobile(data.profiles[5].mobile)
+        .selectDOB(data.profiles[5].DOB.day, data.profiles[5].DOB.month, data.profiles[5].DOB.year)
+        .selectSubjects(data.profiles[5].subjects)
+        .checkHobbies(data.profiles[5].hobbies)
+        .uploadPicture(data.profiles[5].picture)
+        .typeCurrentAddress(data.profiles[5].currentaddress)
+        .selectStateAndCity(data.profiles[5].state, data.profiles[5].city)
+        .clickSubmit();
+      formPage.popup
+        .isNameCorret(data.profiles[5].firstname, data.profiles[5].lastname)
+        .isEmaiCorret(data.profiles[5].email)
+        .isGenderCorret(data.profiles[5].gender)
+        .isMobileCorret(data.profiles[5].mobile)
+        .isDOBCorret(data.profiles[5].DOB.day,data.profiles[5].DOB.month,data.profiles[5].DOB.year)
+        .isSubjectsCorrect(data.profiles[5].subjects)
+        .isHobbiesCorrect(data.profiles[5].hobbies)
+        .isPictureCorrect(data.profiles[5].picture)
+        .isCurrentAddressCorrect(data.profiles[5].currentaddress)
+        .isStateAndCityCorrect(data.profiles[5].state, data.profiles[5].city)
+    });
+  });
+  it("TC-FORM-06: Verify that user can register when user inputs mandatory fields with valid data", ()=>{
+    cy.get("@data").then((data) => {
+      formPage
+        .typeFirstName(data.profiles[6].firstname)
```

Comment:
You can reuse the data in the case above, just ignore the field you don't want to input